cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js

```diff
@@ -451,15 +461,24 @@ class HydratorPlusPlusNodeConfigCtrl {
           output: {},
           numInputStages: 0
         };
+        let hasInputRecords = true;
         angular.forEach(res, (value, key) => {
           if (key !== this.state.node.plugin.label) {
-            if (!this.state.isSource) {
+            if (!this.state.isSource && hasInputRecords) {
               this.previewData.input[key] = this.formatMultipleRecords(value['records.out']);
               this.previewData.numInputStages = Object.keys(res).length;
             }
           } else {
+            let recordsOut = value['records.out'];
             if (!this.state.isSink) {
-              this.previewData.output = this.formatMultipleRecords(value['records.out']);
+              this.previewData.output = this.formatMultipleRecords(recordsOut);
+            }
+            if (_.isEmpty(recordsOut) && previousStageIsCondition) {
```

Comment:
I am not sure if I follow this logic? If the previous stage is a condition node and if it has no output records we set the input records for this to be empty. 

1. Shouldn't that be in the `if` condition block? That is where we are setting the `previewData.input` right?
2. What happens if i have a condition node and another transform connected to my current node and if the condition node has no output records?

Reply:
Addressed 1st item. For 2nd item, the backend doesn't allow this.